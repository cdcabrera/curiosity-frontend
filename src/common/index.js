import { helpers } from './helpers';
import { dateHelpers } from './dateHelpers';
import { downloadHelpers } from './downloadHelpers';

/**
 * @namespace Helpers
 * @property {module} General
 * @property {module} Dates
 * @property {module} Downloads
 */
export { helpers as default, helpers, dateHelpers, downloadHelpers };
