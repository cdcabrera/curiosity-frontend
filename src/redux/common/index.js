import { reduxHelpers } from './reduxHelpers';

/**
 * @memberof Redux State
 * @module Helpers
 * @property {module} ReduxHelpers
 */

export { reduxHelpers as default, reduxHelpers };
