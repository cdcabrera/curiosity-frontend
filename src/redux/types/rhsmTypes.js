const GET_GRAPH_REPORT_CAPACITY_RHSM = 'GET_GRAPH_REPORT_CAPACITY_RHSM';
const GET_HOSTS_INVENTORY_RHSM = 'GET_HOSTS_INVENTORY_RHSM';
const GET_HOSTS_INVENTORY_GUESTS_RHSM = 'GET_HOSTS_INVENTORY_GUESTS_RHSM';
const GET_REPORT_CAPACITY_RHSM = 'GET_REPORT_CAPACITY_RHSM';

/**
 * RHSM API action, reducer types.
 *
 * @type {{GET_GRAPH_REPORT_CAPACITY_RHSM: string, GET_HOSTS_INVENTORY_GUESTS_RHSM: string,
 *     GET_HOSTS_INVENTORY_RHSM: string, GET_REPORT_CAPACITY_RHSM: string}}
 */
const rhsmTypes = {
  GET_GRAPH_REPORT_CAPACITY_RHSM,
  GET_HOSTS_INVENTORY_RHSM,
  GET_HOSTS_INVENTORY_GUESTS_RHSM,
  GET_REPORT_CAPACITY_RHSM
};

export {
  rhsmTypes as default,
  rhsmTypes,
  GET_GRAPH_REPORT_CAPACITY_RHSM,
  GET_HOSTS_INVENTORY_RHSM,
  GET_HOSTS_INVENTORY_GUESTS_RHSM,
  GET_REPORT_CAPACITY_RHSM
};
