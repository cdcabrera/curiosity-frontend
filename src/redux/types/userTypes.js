const GET_API_VERSION = 'GET_API_VERSION';
const USER_AUTH = 'USER_AUTH';
const USER_LOCALE = 'USER_LOCALE';
const USER_LOGOUT = 'USER_LOGOUT';

export { GET_API_VERSION, USER_AUTH, USER_LOCALE, USER_LOGOUT };
