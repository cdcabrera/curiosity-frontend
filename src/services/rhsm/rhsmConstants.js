/**
 * ToDo: rename to RHSM_API_PRODUCT_ID_TYPES
 */
/**
 * RHSM product id type values.
 *
 * @type {{RHEL_ARM: string, SATELLITE: string, OPENSHIFT_DEDICATED: string, RHEL_WORKSTATION: string,
 *     RHEL_COMPUTE_NODE: string, RHEL_X86: string, OPENSHIFT: string, OPENSHIFT_CONTAINER_PLATFORM: string,
 *     SATELLITE_SERVER: string, RHEL_DESKTOP: string, RHEL: string, SATELLITE_CAPSULE: string,
 *     RHEL_SERVER: string, RHEL_IBM_Z: string, RHEL_IBM_POWER: string}}
 */
const RHSM_API_PATH_ID_TYPES = {
  RHEL: 'RHEL',
  RHEL_COMPUTE_NODE: 'RHEL Compute Node',
  RHEL_DESKTOP: 'RHEL Desktop',
  RHEL_SERVER: 'RHEL Server',
  RHEL_WORKSTATION: 'RHEL Workstation',
  RHEL_ARM: 'RHEL for ARM',
  RHEL_IBM_POWER: 'RHEL for IBM Power',
  RHEL_IBM_Z: 'RHEL for IBM z',
  RHEL_X86: 'RHEL for x86',
  OPENSHIFT: 'OpenShift Container Platform',
  OPENSHIFT_CONTAINER_PLATFORM: 'OpenShift Container Platform',
  OPENSHIFT_DEDICATED: 'OpenShift Dedicated',
  SATELLITE: 'Satellite',
  SATELLITE_CAPSULE: 'Satellite Capsule',
  SATELLITE_SERVER: 'Satellite Server'
};

/**
 * RHSM response Error DATA CODE types.
 *
 * @type {{GENERIC: string, OPTIN: string}}
 */
const RHSM_API_RESPONSE_ERROR_CODE_TYPES = {
  GENERIC: 'SUBSCRIPTIONS1003',
  OPTIN: 'SUBSCRIPTIONS1004'
};

/**
 * RHSM API response/query/search parameter SLA type values.
 *
 * @type {{PREMIUM: string, SELF: string, NONE: string, STANDARD: string}}
 */
const RHSM_API_RESPONSE_SLA_TYPES = {
  PREMIUM: 'premium',
  STANDARD: 'standard',
  SELF: 'self-support',
  NONE: ''
};

/**
 * RHSM API response/query/search parameter UOM type values.
 *
 * @type {{CORES: string, SOCKETS: string}}
 */
const RHSM_API_RESPONSE_UOM_TYPES = {
  CORES: 'cores',
  SOCKETS: 'sockets'
};

/**
 * RHSM API response/query/search parameter USAGE type values.
 *
 * @type {{UNSPECIFIED: string, DISASTER: string, DEVELOPMENT: string, PRODUCTION: string}}
 */
const RHSM_API_RESPONSE_USAGE_TYPES = {
  PRODUCTION: 'production',
  DEVELOPMENT: 'development/test',
  DISASTER: 'disaster recovery',
  UNSPECIFIED: ''
};

/**
 * RHSM API query/search parameter of GRANULARITY type values.
 * Schema/map of expected query/search parameter granularity types.
 *
 * @type {{WEEKLY: string, QUARTERLY: string, DAILY: string, MONTHLY: string}}
 */
const RHSM_API_QUERY_GRANULARITY_TYPES = {
  DAILY: 'daily',
  WEEKLY: 'weekly',
  MONTHLY: 'monthly',
  QUARTERLY: 'quarterly'
};

/**
 * RHSM API query/search parameter SORT type values for HOSTS.
 *
 * @type {{CORES: string, HARDWARE: string, SOCKETS: string, MEASUREMENT: string, LAST_SEEN: string, NAME: string}}
 */
const RHSM_API_QUERY_SORT_TYPES = {
  CORES: 'cores',
  HARDWARE: 'hardware_type',
  LAST_SEEN: 'last_seen',
  MEASUREMENT: 'measurement_type',
  NAME: 'display_name',
  SOCKETS: 'sockets'
};

/**
 * RHSM API query/search parameter SORT type values for SUBSCRIPTIONS.
 *
 * @type {{UOM: string, PHYSICAL_CAPACITY: string, USAGE: string, UPCOMING_EVENT_DATE: string,
 *     VIRTUAL_CAPACITY: string, TOTAL_CAPACITY: string, SKU: string, PRODUCT_NAME: string,
 *     SERVICE_LEVEL: string}}
 */
const RHSM_API_QUERY_SUBSCRIPTIONS_SORT_TYPES = {
  SKU: 'sku',
  PRODUCT_NAME: 'product_name',
  SERVICE_LEVEL: 'service_level',
  USAGE: 'usage',
  UPCOMING_EVENT_DATE: 'upcoming_event_date',
  PHYSICAL_CAPACITY: 'physical_capacity',
  VIRTUAL_CAPACITY: 'virtual_capacity',
  TOTAL_CAPACITY: 'total_capacity',
  UOM: 'uom'
};

/**
 * RHSM API query/search parameter SORT DIRECTION type values.
 *
 * @type {{ASCENDING: string, DESCENDING: string}}
 */
const RHSM_API_QUERY_SORT_DIRECTION_TYPES = {
  ASCENDING: 'asc',
  DESCENDING: 'desc'
};

/**
 * RHSM API response/query/search parameter SLA type values.
 */
const RHSM_API_QUERY_SLA_TYPES = RHSM_API_RESPONSE_SLA_TYPES;

/**
 * RHSM API response/query/search parameter UOM type values.
 */
const RHSM_API_QUERY_UOM_TYPES = RHSM_API_RESPONSE_UOM_TYPES;

/**
 * RHSM API response/query/search parameter USAGE type values.
 */
const RHSM_API_QUERY_USAGE_TYPES = RHSM_API_RESPONSE_USAGE_TYPES;

/**
 * RHSM API query/search parameter OPTIN type values.
 *
 * @type {{TALLY_SYNC: string, TALLY_REPORT: string, CONDUIT_SYNC: string}}
 */
const RHSM_API_QUERY_SET_OPTIN_TYPES = {
  CONDUIT_SYNC: 'enable_conduit_sync',
  TALLY_REPORT: 'enable_tally_reporting',
  TALLY_SYNC: 'enable_tally_sync'
};

// ToDo: rename to ...TALLY_CAPACITY_TYPES
/**
 * RHSM API query/search parameter CAPACITY type values.
 *
 * @type {{GRANULARITY: string, USAGE: string, END_DATE: string, SLA: string, START_DATE: string}}
 */
const RHSM_API_QUERY_SET_REPORT_CAPACITY_TYPES = {
  END_DATE: 'ending',
  GRANULARITY: 'granularity',
  SLA: 'sla',
  START_DATE: 'beginning',
  USAGE: 'usage'
};

// ToDo: rename to ...SET_HOST_REPORT_TYPES
/**
 * RHSM API query/search parameter INVENTORY type values.
 *
 * @type {{UOM: string, USAGE: string, DIRECTION: string, SORT: string, OFFSET: string,
 *     SLA: string, LIMIT: string}}
 */
const RHSM_API_QUERY_SET_INVENTORY_TYPES = {
  DIRECTION: 'dir',
  DISPLAY_NAME: 'display_name_contains',
  LIMIT: 'limit',
  OFFSET: 'offset',
  SLA: 'sla',
  SORT: 'sort',
  UOM: 'uom',
  USAGE: 'usage'
};

// ToDo: rename to ...SET_HYPERVISOR_GUEST_REPORT_TYPES
/**
 * RHSM API query/search parameter GUESTS INVENTORY type values.
 *
 * @type {{OFFSET: string, LIMIT: string}}
 */
const RHSM_API_QUERY_SET_INVENTORY_GUESTS_TYPES = {
  LIMIT: 'limit',
  OFFSET: 'offset'
};

// ToDo: rename to ...SET_SUBSCRIPTIONS_REPORT_TYPES
/**
 * RHSM API query/search parameter SUBSCRIPTIONS INVENTORY type values.
 */
const RHSM_API_QUERY_SET_INVENTORY_SUBSCRIPTIONS_TYPES = {
  ...RHSM_API_QUERY_SET_INVENTORY_TYPES
};

/**
 * RHSM API query/search parameter values.
 */
const RHSM_API_QUERY_TYPES = {
  ...RHSM_API_QUERY_SET_OPTIN_TYPES,
  ...RHSM_API_QUERY_SET_REPORT_CAPACITY_TYPES,
  ...RHSM_API_QUERY_SET_INVENTORY_TYPES,
  ...RHSM_API_QUERY_SET_INVENTORY_GUESTS_TYPES,
  ...RHSM_API_QUERY_SET_INVENTORY_SUBSCRIPTIONS_TYPES
};

const rhsmConstants = {
  RHSM_API_PATH_ID_TYPES,
  RHSM_API_QUERY_GRANULARITY_TYPES,
  RHSM_API_QUERY_SORT_TYPES,
  RHSM_API_QUERY_SUBSCRIPTIONS_SORT_TYPES,
  RHSM_API_QUERY_SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_SLA_TYPES,
  RHSM_API_QUERY_UOM_TYPES,
  RHSM_API_QUERY_USAGE_TYPES,
  RHSM_API_QUERY_SET_OPTIN_TYPES,
  RHSM_API_QUERY_SET_REPORT_CAPACITY_TYPES,
  RHSM_API_QUERY_SET_INVENTORY_TYPES,
  RHSM_API_QUERY_SET_INVENTORY_GUESTS_TYPES,
  RHSM_API_QUERY_SET_INVENTORY_SUBSCRIPTIONS_TYPES,
  RHSM_API_QUERY_TYPES,
  RHSM_API_RESPONSE_ERROR_CODE_TYPES,
  RHSM_API_RESPONSE_SLA_TYPES,
  RHSM_API_RESPONSE_UOM_TYPES,
  RHSM_API_RESPONSE_USAGE_TYPES
  // RHSM_API_SLA_TYPES,
  // RHSM_API_UOM_TYPES,
  // RHSM_API_USAGE_TYPES,
  // SUBSCRIPTIONS_REPORT_SERVICE_LEVEL,
  // SUBSCRIPTIONS_REPORT_UOM,
  // SUBSCRIPTIONS_REPORT_USAGE
};

export {
  rhsmConstants as default,
  rhsmConstants,
  RHSM_API_PATH_ID_TYPES,
  RHSM_API_QUERY_GRANULARITY_TYPES,
  RHSM_API_QUERY_SORT_TYPES,
  RHSM_API_QUERY_SUBSCRIPTIONS_SORT_TYPES,
  RHSM_API_QUERY_SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_SLA_TYPES,
  RHSM_API_QUERY_UOM_TYPES,
  RHSM_API_QUERY_USAGE_TYPES,
  RHSM_API_QUERY_SET_OPTIN_TYPES,
  RHSM_API_QUERY_SET_REPORT_CAPACITY_TYPES,
  RHSM_API_QUERY_SET_INVENTORY_TYPES,
  RHSM_API_QUERY_SET_INVENTORY_GUESTS_TYPES,
  RHSM_API_QUERY_SET_INVENTORY_SUBSCRIPTIONS_TYPES,
  RHSM_API_QUERY_TYPES,
  RHSM_API_RESPONSE_ERROR_CODE_TYPES,
  RHSM_API_RESPONSE_SLA_TYPES,
  RHSM_API_RESPONSE_UOM_TYPES,
  RHSM_API_RESPONSE_USAGE_TYPES
};
