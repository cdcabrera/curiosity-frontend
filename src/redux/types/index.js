import { appTypes } from './appTypes';
import { graphTypes } from './graphTypes';
import { inventoryTypes } from './inventoryTypes';
import { platformTypes } from './platformTypes';
import { queryTypes } from './queryTypes';
import { rhsmTypes } from './rhsmTypes';
import { toolbarTypes } from './toolbarTypes';
import { userTypes } from './userTypes';

/**
 * @memberof Redux State
 * @module Types
 * @property {module} AppTypes
 * @property {module} GraphTypes
 * @property {module} InventoryTypes
 * @property {module} PlatformTypes
 * @property {module} QueryTypes
 * @property {module} RhsmTypes
 * @property {module} ToolbarTypes
 * @property {module} UserTypes
 */

/**
 * Redux types
 *
 * @type {{app: {STATUS_4XX: string, STATUS_5XX: string}, toolbar: {SET_FILTER_TYPE: string, SET_ACTIVE_FILTERS: string},
 *     rhsm: {GET_GRAPH_REPORT_CAPACITY_RHSM: string, GET_MESSAGE_REPORTS_RHSM: string, GET_HOSTS_INVENTORY_GUESTS_RHSM: string,
 *     GET_GRAPH_CAPACITY_RHSM: string, GET_SUBSCRIPTIONS_INVENTORY_RHSM: string, GET_HOSTS_INVENTORY_RHSM: string,
 *     GET_INSTANCES_INVENTORY_RHSM: string, GET_GRAPH_TALLY_RHSM: string},
 *     query: {SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES: object,
 *     SET_QUERY_RHSM_TYPES: object, SET_QUERY_RHSM_GUESTS_INVENTORY_TYPES: object, SET_QUERY_CLEAR: string,
 *     SET_QUERY_CLEAR_INVENTORY_LIST: string, SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES: object, SET_QUERY: string,
 *     SET_QUERY_RESET_INVENTORY_LIST: string, SET_QUERY_CLEAR_INVENTORY_GUESTS_LIST: string},
 *     inventory: {CLEAR_INVENTORY_GUESTS: string, SET_INVENTORY_TAB: string}, user: {USER_LOGOUT: string, USER_LOCALE: string,
 *     GET_USER_OPTIN: string, UPDATE_USER_OPTIN: string, DELETE_USER_OPTIN: string}, graph: {SET_GRAPH_LEGEND: string},
 *     platform: {PLATFORM_APP_NAME: string, PLATFORM_USER_AUTH: string, PLATFORM_GLOBAL_FILTER_HIDE: string,
 *     PLATFORM_INIT: string, PLATFORM_SET_NAV: string, PLATFORM_CLEAR_NOTIFICATIONS: string, PLATFORM_ADD_NOTIFICATION: string,
 *     PLATFORM_REMOVE_NOTIFICATION: string, PLATFORM_ON_NAV: string}}}
 */
const reduxTypes = {
  app: appTypes,
  graph: graphTypes,
  inventory: inventoryTypes,
  platform: platformTypes,
  query: queryTypes,
  rhsm: rhsmTypes,
  toolbar: toolbarTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  appTypes,
  graphTypes,
  inventoryTypes,
  platformTypes,
  queryTypes,
  rhsmTypes,
  toolbarTypes,
  userTypes
};
