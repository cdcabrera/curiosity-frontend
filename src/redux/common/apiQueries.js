import { rhsmApiTypes } from '../../types/rhsmApiTypes';
import { reduxHelpers } from './reduxHelpers';

/**
 * Parse a query object against a schema for specific RHSM endpoints.
 *
 * @param {object} query
 * @returns {{query: {}, guestsQuery: {}, graphQuery: {}, inventoryQuery: {}, toolbarQuery: {}}}
 */
const parseRhsmQuery = (query = {}) => {
  const graphQuery = reduxHelpers.setApiQuery(query, rhsmApiTypes.RHSM_API_QUERY_SET_REPORT_CAPACITY_TYPES);
  const guestsQuery = reduxHelpers.setApiQuery(query, rhsmApiTypes.RHSM_API_QUERY_SET_INVENTORY_GUESTS_TYPES);
  const inventoryQuery = reduxHelpers.setApiQuery(query, rhsmApiTypes.RHSM_API_QUERY_SET_INVENTORY_TYPES);
  const pagingQuery = reduxHelpers.setApiQuery(query, rhsmApiTypes.RHSM_API_QUERY_SET_PAGING_TYPES);

  return {
    query,
    graphQuery,
    guestsQuery,
    inventoryQuery,
    pagingQuery,
    toolbarQuery: query
  };
};

const apiQueries = {
  parseRhsmQuery
};

export { apiQueries as default, apiQueries };
