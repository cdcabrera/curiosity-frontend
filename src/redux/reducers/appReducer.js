import _get from 'lodash/get';
import { appTypes, platformTypes } from '../types';
import { rhsmConstants } from '../../services/rhsm/rhsmConstants';
import { reduxHelpers } from '../common';

/**
 * Application related state reducer.
 *
 * @memberof Reducers
 * @module AppReducer
 */

/**
 * Initial state.
 *
 * @private
 * @type {{apiErrors: {}, auth: {}, exports: {}, exportsExisting: {}, optin: {}, locale: {}, errors: {}}}
 */
const initialState = {
  apiErrors: {},
  auth: {},
  errors: {},
  exports: {},
  exportsExisting: {},
  locale: {},
  optin: {}
};

/**
 * Apply application observer/reducer logic to state, against actions.
 *
 * @param {object} state
 * @param {object} action
 * @returns {object|{}}
 */
const appReducer = (state = initialState, action) => {
  switch (action.type) {
    case reduxHelpers.HTTP_STATUS_RANGE(appTypes.STATUS_4XX):
      const action4xxStatus = reduxHelpers.getStatusFromResults(action);

      if (action4xxStatus === 401 || action4xxStatus === 403) {
        const errorCodes = _get(reduxHelpers.getDataFromResults(action), [rhsmConstants.RHSM_API_RESPONSE_ERRORS], []);

        return reduxHelpers.setStateProp(
          'errors',
          {
            error: true,
            errorMessage: reduxHelpers.getMessageFromResults(action),
            data: errorCodes.map(value => value[rhsmConstants.RHSM_API_RESPONSE_ERRORS_TYPES.CODE]),
            status: reduxHelpers.getStatusFromResults(action)
          },
          {
            state,
            initialState
          }
        );
      }

      return state;
      /*
    case '>>> COMPONENT API':
      if (action.error === true) {
        return reduxHelpers.setStateProp(
          'apiErrors',
          {
            estimatedApiCalls: action.estimatedApiCalls,

          },
          {
            state,
            initialState
          }
        );
      }
      return state;
      */
      /*
    case reduxHelpers.HTTP_STATUS_RANGE(appTypes.STATUS_5XX):
      const action5xxStatus = reduxHelpers.getStatusFromResults(action);

      if (action5xxStatus >= 500) {
        return reduxHelpers.setStateProp(
          'apiErrors',
          {
            [action.meta.productId || 'all']: {
              error: true,
              messages: [
                ...(state.apiErrors?.[action.meta.productId || 'all']?.messages || []),
                {
                  meta: action.meta,
                  message: reduxHelpers.getMessageFromResults(action),
                  status: reduxHelpers.getStatusFromResults(action)
                }
              ]
            }
          },
          {
            state,
            initialState
          }
        );
      }

      return state;
       */
    case platformTypes.SET_PLATFORM_EXPORT_STATUS:
      return reduxHelpers.setStateProp(
        'exports',
        {
          [action.id]: {
            ...action,
            pending: [
              // Only a selected format updates/reuses the pending list
              ...((action.isSelectUpdated === true && state?.exports?.[action.id]?.pending) || []),
              ...((Array.isArray(action.pending) && action.pending) || (action.pending && [action.pending]) || [])
            ]
          }
        },
        {
          state,
          reset: false
        }
      );
    case platformTypes.SET_PLATFORM_EXPORT_RESET:
      return reduxHelpers.setStateProp(
        null,
        {
          ...state,
          exports: {},
          exportsExisting: {}
        },
        {
          state,
          initialState
        }
      );
    default:
      return reduxHelpers.generatedPromiseActionReducer(
        [
          { ref: 'locale', type: appTypes.USER_LOCALE },
          { ref: 'optin', type: [appTypes.DELETE_USER_OPTIN, appTypes.GET_USER_OPTIN, appTypes.UPDATE_USER_OPTIN] },
          { ref: 'exportsExisting', type: platformTypes.SET_PLATFORM_EXPORT_EXISTING_STATUS },
          { ref: 'auth', type: platformTypes.PLATFORM_USER_AUTH }
        ],
        state,
        action
      );
  }
};

appReducer.initialState = initialState;

export { appReducer as default, initialState, appReducer };
