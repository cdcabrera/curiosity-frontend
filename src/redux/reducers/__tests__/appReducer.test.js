import appReducer from '../appReducer';
import { rhsmConstants } from '../../../services/rhsm/rhsmConstants';
import { appTypes as types } from '../../types';
import { reduxHelpers } from '../../common';

describe('UserReducer', () => {
  it('should return the initial state', () => {
    expect(appReducer.initialState).toBeDefined();
  });

  it('should handle specific http status types', () => {
    const specificTypes = [
      { type: types.STATUS_4XX, status: 400 },
      { type: types.STATUS_4XX, status: 401 },
      { type: types.STATUS_4XX, status: 403 }
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: reduxHelpers.HTTP_STATUS_RANGE(value.type),
        error: true,
        payload: {
          message: `Request failed with status code ${value.status}`,
          response: {
            status: value.status,
            statusText: 'Error',
            data: {
              [rhsmConstants.RHSM_API_RESPONSE_ERRORS]: [
                {
                  [rhsmConstants.RHSM_API_RESPONSE_ERRORS_TYPES.CODE]:
                    rhsmConstants.RHSM_API_RESPONSE_ERRORS_CODE_TYPES.OPTIN
                }
              ]
            }
          }
        }
      };

      const resultState = appReducer(undefined, dispatched);

      expect({ type: value.type, result: resultState }).toMatchSnapshot(`http status ${value.type} ${value.status}`);
    });
  });
});
