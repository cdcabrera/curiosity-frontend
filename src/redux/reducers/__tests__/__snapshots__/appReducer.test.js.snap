// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReducer should handle specific http status types: http status 4XX 400 1`] = `
{
  "result": {
    "errors": {},
  },
  "type": "4XX",
}
`;

exports[`UserReducer should handle specific http status types: http status 4XX 401 1`] = `
{
  "result": {
    "errors": {
      "data": [
        "SUBSCRIPTIONS1004",
      ],
      "error": true,
      "errorMessage": "Request failed with status code 401",
      "status": 401,
    },
  },
  "type": "4XX",
}
`;

exports[`UserReducer should handle specific http status types: http status 4XX 403 1`] = `
{
  "result": {
    "errors": {
      "data": [
        "SUBSCRIPTIONS1004",
      ],
      "error": true,
      "errorMessage": "Request failed with status code 403",
      "status": 403,
    },
  },
  "type": "4XX",
}
`;
